File: 46. Permutations/Solution.cpp
class Solution {
public:
    void perm(vector<int> v,int a,vector<vector<int>> &re){
        if(v.size()==a){
            re.push_back(v);
            return;
        }
        for(int i=a; i<v.size(); i++){
            swap(v[a],v[i]);
            perm(v,a+1,re);
            swap(v[a],v[i]);
        }
    }
    vector<vector<int>> permute(vector<int>& nums) {
        vector<vector<int>> re;
        perm(nums,0,re);
        return re;
    }
};

File: 46. Permutations/Solution.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        def perm(a:int,re:List[List[int]])->None:
            if len(nums)==a:
                re.append(nums[:])
                return
            for i in range(a,len(nums)):
                nums[a],nums[i]=nums[i],nums[a]
                perm(a+1,re)
                nums[a],nums[i]=nums[i],nums[a]
        re = []
        perm(0,re)
        return re
